client/dom/radiobutton.ts

```diff
@@ -0,0 +1,104 @@
+/* makes radio buttons */
+
+type RadioButtonOpts = {
+	/** Fires .oninput. Intended for general use. */
+	callback?: (f?: any) => void
+	/** Required.  */
+	holder: any
+	/** common Name of <input>, use random number if not given */
+	inputName?: string | number
+	/** Mass ui specific logic. Optional callback methods for
+	 * non oninput events. Intended to address needs for assistive techology
+	 */
+	listeners?: {
+		/** Fires on onmouseup and onkeyup for the button and text label
+		 */
+		input: () => void
+	}
+	/** arr of objs defining the radio buttons and properties */
+	options: OptionEntry[]
+	/** css to be applied to each <div> of the options
+	 * e.g. { "padding":"5px", "display":"inline-block" }
+	 */
+	styles?: { [index: string]: string | number }
+}
+
+type OptionEntry = {
+	/** only set for only *one* option */
+	checked?: boolean | number
+	/** Text shown in the span to the right of the radio button */
+	label: string
+	/** Text shown in tooltip */
+	title?: string
+	/** Should correspond to 'currValue' in callbacks */
+	value: string | number
+}
+
+let nameSuffix = 0
+
+export function make_radios(opts: RadioButtonOpts) {
+	if (!opts.callback && !opts.listeners) throw `Missing event callback for radios [#dom/radiobutton.js]`
+	if (opts.callback && opts.listeners)
+		throw `Both callback() and .listeners defined [#dom/radiobutton.js]. Only supply one.`
+	const inputName = opts.inputName || 'pp-dom-radio-' + nameSuffix++
+
+	const divs = opts.holder
+		.selectAll('div')
+		.style('display', 'block')
+		.data(opts.options, (d: any) => d.value)
+
+	const labels = divs
+		.enter()
+		.append('div')
+		.attr('aria-label', (d: OptionEntry) => d.title)
+		.style('display', opts.styles && 'display' in opts.styles ? opts.styles.display : 'block')
+		.style('padding', opts.styles && 'padding' in opts.styles ? opts.styles.padding : '5px')
+		.append('label')
+
+	if (opts.styles) {
+		for (const k in opts.styles) {
+			labels.style(k, opts.styles[k])
+		}
+	}
+
+	const inputs = labels
+		.append('input')
+		.attr('type', 'radio')
+		.attr('name', inputName)
+		.attr('value', (d: OptionEntry) => d.value)
+		.style('vertical-align', 'top')
+		.style('margin-top', '2px')
+		.style('margin-right', 0)
+		.property('checked', (d: OptionEntry) => d?.checked)
+	if (opts.callback) {
+		inputs.on('input', async (event: KeyboardEvent, d: OptionEntry) => {
+			inputs.property('disabled', true)
+			if (!opts.callback) return //So eslint doesn't complain
+			await opts.callback(d.value)
```

Comment:
should explain why disabling and re-enabling is required before and after calling the callback()

Reply:
also, it seems like `radio['currValue']` should be updated here, to equal `d.value`.